rpc/errors_test.go

```diff
@@ -0,0 +1,24 @@
+package rpc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/test-go/testify/require"
+)
+
+func TestRPCError(t *testing.T) {
+	if testEnv == "mock" {
+		testConfig := beforeEach(t)
+		_, err := testConfig.provider.ChainID(context.Background())
+		require.NoError(t, err)
+		require.Nil(t, err)
```

Comment:
One of these checks are redundant: if return type is ptr to type you can just leave only `require.Nil` if it's an `error` - leave only `require.NoError(...)`